src/dali/plugin/input/csv/nexo.py

```diff
@@ -97,53 +82,63 @@ def load(self) -> List[AbstractTransaction]:
+                    # if it is non-zero, we use it, otherwise we use unknown as the value
+                    # the spot price contains $ char, so we remove it
+                    spot_price = RP2Decimal(re.sub(r'[^\d.]', '', line[self.__SPOT_PRICE_INDEX]))
+
+                    if spot_price.is_zero():
+                        spot_price = Keyword.UNKNOWN.value
+                    else:
+                        spot_price = str(spot_price)
+
                     result.append(
-                        InTransaction(
-                            plugin=self.__NEXO,
-                            unique_id=transaction_id,
-                            raw_data=raw_data,
-                            exchange=self.__NEXO,
-                            holder=self.account_holder,
-                            timestamp=timestamp_with_timezone,
-                            asset=currency,
-
-                            transaction_type=Keyword.INTEREST.name,
-                            # nexo does give us the spot price, but it's often 0 if subcent
-                            spot_price=Keyword.UNKNOWN.value,
-                            crypto_in=amount,
-                            fiat_fee="0",
-                        )
+                        InTransaction(**(common_params | {
+                            'exchange':self.__NEXO,
+                            'holder':self.account_holder,
+
+                            'transaction_type':Keyword.INTEREST.name,
+                            'spot_price': spot_price,
+                            'crypto_in': amount,
+                            'fiat_fee': "0",
+                        }))
                     )
                 elif transaction_type in [_LOCKING_TERM_DEPOSIT, _UNLOCKING_TERM_DEPOSIT]:
+                    # These are unique to Nexo thing: they "lock" your crypto in a "fixed term" deposit which earns higher interest.
+                    # i.e. these transactions just indicate that you cannot withdraw these funds while these are locked. So they effect your available balance.
                     # I don't think we need to record locking/unlocking deposits for term interest
                     self.__logger.debug("Skipping lock or unlock deposit: %s", line)
                 elif transaction_type == _DEPOSIT:
                     result.append(
-                        IntraTransaction(
-                            plugin=self.__NEXO,
-                            unique_id=transaction_id,
-                            raw_data=raw_data,
-                            timestamp=timestamp_with_timezone,
-                            asset=currency,
-
-                            crypto_received=amount,
+                        IntraTransaction(**(common_params | {
+                            'crypto_received':amount,
 
                             # most likely, it's you, but we can't say for sure
```

Comment:
Cleared this up. I think we should keep the comment—for new devs coming to the codebase it makes it easier to understand _why_ we are using unknown, otherwise it seems like an error/omission.